Restore errors and warnings (#67803)
openshift/openshift-docs: modules/migration-debugging-velero-resources.adoc

  backup describe 0e44ae00-5dc3-11eb-9ca8-df7e5254778b-2d8ql
----

> The following types of restore errors and warnings are shown in the output of a `velero describe` request:

Reviewer: > The following types of restore errors and warnings are shown in the output of a `velero describe` request:

Do we have a section for backup as well?

Reviewer: I guess this PR is scoped to restore only..

* `Velero`: A list of messages related to the operation of Velero itself, for example, messages related to connecting to the cloud, reading a backup file, and so on
* `Cluster`: A list of messages related to backing up or restoring cluster-scoped resources
* `Namespaces`: A list of list of messages related to backing up or restoring resources stored in namespaces

One or more errors in one of these categories results in a `Restore` operation receiving the status of `PartiallyFailed` and not `Completed`. Warnings do not lead to a change in the completion status.

[IMPORTANT]
====
* For resource-specific errors, that is, `Cluster` and `Namespaces` errors, the `restore describe --details` output includes a resource list that lists all resources that Velero succeeded in restoring. For any resource that has such an error, check to see if the resource is actually in the cluster.

* If there are `Velero` errors, but no resource-specific errors, in the output of a `describe` command, it is possible that the restore completed without any actual problems in restoring workloads, but carefully validate post-restore applications.
+

> For example, if the output contains `PodVolumeRestore` or Node Agent-related errors, check the status of `PodVolumeRestores` and `DataDownloads`. If none of these are failed or still running, then volume data might have been fully restored.

Reviewer 2: Should this be node agent related errors?

Author: @JoeAldinger Good question -- let me check my sources.

Author: @JoeAldinger I'll  change that -- good catch.

> For example, if the output contains `PodVolumeRestore` or node sgent-related errors, check the status of `PodVolumeRestores` and `DataDownloads`. If none of these are failed or still running, then volume data might have been fully restored.

Reviewer 2: @RichardHoch can you fix this typo in a follow up PR please?

Author: On it.

  backup describe 0e44ae00-5dc3-11eb-9ca8-df7e5254778b-2d8ql
----

The following types of restore errors and warnings are shown in the output of a `velero describe` request:

> * `Velero`: A list of messages related to the operation of Velero itself (for example, messages related to connecting to the cloud, reading a backup file, etc.)
> * `Cluster`: A list of messages related to backup or restore of cluster-scoped resources
> * `Namespaces`: A list of list of messages related to backup or restore namespace-scoped resources

Reviewer 3: ```suggestion
* `Velero`: A list of messages related to the operation of Velero itself, for example, messages related to the connection to the cloud, reading a backup file, and so on.
* `Cluster`: A list of messages related to the backup or restore of cluster resources
* `Namespaces`: A list of messages related to the backup or restore of resources stored in namespaces
```

> * `Namespaces`: A map of namespace name to slice of messages related to backup or restore namespace-scoped resources

Reviewer: > Anyway...I'll change "slice" to "list" ...

It still says slice here.

https://redhat-internal.slack.com/archives/C0144ECKUJ0/p1707406131362059?thread_ts=1707388097.633329&cid=C0144ECKUJ0

Reviewer: ```suggestion
* `Velero`: A list of messages related to the operation of Velero itself (for example, messages related to connecting to the cloud, reading a backup file, etc.)
* `Cluster`: A list of messages related to backup or restore of cluster-scoped resources
* `Namespaces`: A list of list of messages related to backup or restore namespace-scoped resources
```

> One or more errors in any of the above categories causes a `Restore` operation to have a status of `PartiallyFailed`, rather than `Completed`. Warnings do not cause a change to the completion status.

Reviewer 3: ```suggestion
One or more errors in one of these categories results in a `Restore` operation receiving the status `PartiallyFailed` and not `Completed`. Warnings do not lead to a change in the completion status.
```

> [NOTE]
> ====
> For resource-specific errors, that is, `Cluster` and `Namespaces` errors, the `restore describe --details` output includes a resource list that lists all resources that Velero succeeded in restoring. For any resource that has such an error, check to see if the resource is actually in the cluster.
> ====
> 
> [NOTE]
> ====
> If there are `Velero` errors, but no resource-specific errors, in the output of a `describe` command, it is possible that the restore completed without any actual problems in restoring workloads, but carefully validate post-restore applications.
> For example, if the output contains `PodVolumeRestore` or Node Agent-related errors, check the status of `PodVolumeRestores` and `DataDownloads`. If none of these are failed or still running, then volume data might have been fully restored.
> ====
> 

Reviewer 3: ```suggestion
[IMPORTANT]
====
* For resource-specific errors, such as `Cluster` and `Namespaces` errors, the `restore describe --details` output contains a resource list that lists all resources that Velero was able to restore successfully. For each resource that shows such an error, check whether the resource is actually in the cluster.

* If the output of a description command contains `Velero` errors but no resource-specific errors, it is possible that the restore completed without any actual workload restore issues, but check the applications carefully after restoration.
For example, if the output contains `PodVolumeRestore` or Node Agent related errors, check the status of `PodVolumeRestores` and `DataDownloads`. If none of these applications have failed or are still running, the volume data might have been fully restored.
====
```

Reviewer 3: Clubbing the two seprate notes together. Since, there is action involved, I would suggest changing it into an "Important" notice.
